common/src/main/java/software/tnb/common/utils/IOUtils.java

```diff
@@ -32,17 +33,30 @@ private IOUtils() {
 
     public static void writeFile(Path file, String content) {
         try {
-            Path dir = file.getParent();
-            if (!Files.exists(dir)) {
-                LOG.debug("Creating directory " + dir);
-                Files.createDirectory(dir);
-            }
+            checkIfExistAndCreate(file.getParent());
             Files.write(file, content.getBytes());
         } catch (IOException e) {
             throw new RuntimeException("Unable to write to " + file, e);
         }
     }
 
+    public static void copyFile(Path resFolder, String source, String target) throws IOException {
+        try {
+            checkIfExistAndCreate(resFolder.resolve(target).getParent());
+            Files.copy(Paths.get(source), resFolder.resolve(target),
+                StandardCopyOption.REPLACE_EXISTING);
+        } catch (IOException e) {
+            throw new RuntimeException("Unable to copy to " + source, e);
+        }
+    }
+
+    private static void checkIfExistAndCreate(Path dir) throws IOException {
+        if (!Files.exists(dir)) {
+            LOG.debug("Creating directory " + dir);
+            Files.createDirectory(dir);
```

Comment:
if you use `Files.createDirectories` instead of `Files.createDirectory` the full tree will be created even if the parent folder does not exist, without exceptions. IMO all the method `checkIfExistAndCreate` can be replaced with `Files.createDirectories` that it does exactly this [API Doc](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/Files.html#createDirectories(java.nio.file.Path,java.nio.file.attribute.FileAttribute...))